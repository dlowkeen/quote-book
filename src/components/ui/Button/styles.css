/* Button Styles */

/* TODO - mixin pattern so we only need a single button class? */
.btn {
  /* resets */
  outline: none;
  border: none;
  background: none;
  text-transform: none; /* only necessary for anchor buttons -- move to subclass if we split those off into a separate compontent */

  /* shared button styles */
  border-radius: $uiBorderRadius;
  cursor: pointer;
  letter-spacing: 1px;
  min-width: 18rem;
  padding: 0.5em 2em;
}

.btn--primary {
  background: $colorCurrencyGreen;
  border: 1px solid $colorCurrencyGreen;
  color: $colorTypeLight;

  &[disabled] {
    cursor: default;
    opacity: 0.5;
  }
}

.btn--secondary {
  background: transparent;
  border: 1px solid $colorCurrencyGreen;
  color: $colorCurrencyGreen;

  &[disabled] {
    border-color: $colorGrey2;
    color: $colorGrey2;
    cursor: default;
  }
}
