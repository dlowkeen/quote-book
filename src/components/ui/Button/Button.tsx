import React from 'react';

import * as styles from './styles.css';

interface IButtonClassByType {
  primary: string;
  secondary: string;
  [key: string]: string;
}

interface IButtonProps {
  onClick: (e: any) => void;
  type: string;
  title: string;
}

const buttonClassByType: IButtonClassByType = {
  primary: styles.btnPrimary,
  secondary: styles.btnSecondary,
};

const Button: React.FunctionComponent<IButtonProps> = ({
  type,
  onClick,
  title,
}) => (
  <button
    className={`${styles.btn} ${buttonClassByType[type]}`}
    onClick={onClick}
  >
    {title}
  </button>
);

export default Button;
