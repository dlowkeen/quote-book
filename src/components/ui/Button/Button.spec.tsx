import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('The Button component', () => {
  it('renders as expected', () => {
    const tree = renderer
      .create(
        <Button
          type='primary'
          title='you know you wanna...'
          onClick={() => 'clickedy-doo-daa'}
        />,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
