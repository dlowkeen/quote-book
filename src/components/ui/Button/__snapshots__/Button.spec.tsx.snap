// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component renders as expected 1`] = `
<button
  className="styles__btn__1qKLL styles__btn--primary__1h19K"
  onClick={[Function]}
>
  you know you wanna...
</button>
`;
